### ISSUE TITLE
sidebar generation does not work as expected when nbs are in folders 3 or more levels deep

### ISSUE THREAD

**Issue by deven-gqc** [AnswerDotAI/nbdev#1357]
# Provide a minimally reproducible example

1. start a new project with the `nbdev_new` template
2. add new notebooks in the `nbs` folder 3 or more levels deep

This is what a project of mine look like

![image](https://github.com/fastai/nbdev/assets/86305049/6a62e10f-4e33-4eaf-8b63-e907fbdc29cd)

Upon running `nbdev_preview`, the sidebar is generated but it created a different section of a nb in the same category

![image](https://github.com/fastai/nbdev/assets/86305049/495355d7-19c0-4912-b2d5-31973079e5bc)

The generated sidebar looks like this,

```yml
website:
  sidebar:
    contents:
      - index.ipynb
      - section: exp
        contents:
          - section: camvid
            contents:
              - exp/camvid/4 labels.ipynb
      - section: exp
        contents:
          - section: camvid
            contents:
              - exp/camvid/Untitled0.ipynb
      - section: exp
        contents:
          - section: camvid
            contents:
              - exp/camvid/camvid.ipynb
      - section: exp
        contents:
          - section: camvid
            contents:
              - exp/camvid/skimage save camvid labels.ipynb

```

It should ideally look like this
```yml
website:
  sidebar:
    contents:
      - index.ipynb
      - section: exp
        contents:
          - section: camvid
            contents:
              - exp/camvid/4 labels.ipynb
              - exp/camvid/Untitled0.ipynb
              - exp/camvid/camvid.ipynb
              - exp/camvid/skimage save camvid labels.ipynb
```

@hamelsmu do you think this is expected behavior or is it a bug?

**Comment by hamelsmu** [comment]
Seems like a bug - won't have time to look at this for a long time though :/ 

If you are up for please feel free to send a PR and I'll happily review

**Comment by deven367** [comment]
Fixed in #1307. @hamelsmu can you please close this issue? this is now fixed.
